Framework/Algorithms/src/CalculateFlatBackground.cpp

```diff
@@ -157,11 +206,18 @@ void CalculateFlatBackground::exec() {
       double variance = -1;
 
       // Now call the function the user selected to calculate the background
-      const double background =
-          std::string(getProperty("mode")) == "Mean"
-              ? this->Mean(outputWS, currentSpec, startX, endX, variance)
-              : this->LinearFit(outputWS, currentSpec, startX, endX);
-
+      double background = -1;
+      switch (mode) {
+      case Modes::LINEAR_FIT:
+        background = LinearFit(outputWS, currentSpec, startX, endX);
+        break;
+      case Modes::MEAN:
+        background = Mean(outputWS, currentSpec, startX, endX, variance);
+        break;
+      case Modes::MOVING_AVERAGE:
+        background = movingAverage(outputWS, currentSpec, windowWidth);
+        break;
+      }
       if (background < 0) {
         g_log.warning() << "Problem with calculating the background number of "
                            "counts spectrum with index " << currentSpec
```

Comment:
Could this warning be more specific? When I saw it in the log window I couldn't understand what was going on.